src/MqttPublisher.h

```diff
@@ -50,10 +51,19 @@ class MqttPublisher
       client.setCredentials(config.username, config.password);
     }
     client.setCleanSession(true);
-    client.setWill(lastWillTopic.c_str(), MQTT_LWT_QOS, MQTT_LWT_RETAIN, MQTT_LWT_PAYLOAD_OFFLINE);
+    if(config.jsonPayload[0] == 's')
+    {
+      const char* willTopic = lastWillTopic.c_str();
+      const char* buf = new_json_wrap(willTopic, MQTT_LWT_PAYLOAD_OFFLINE);
+      client.setWill(willTopic, MQTT_LWT_QOS, MQTT_LWT_RETAIN, buf);
```

Comment:
While working on the latest changes (including the replacement of the MQTT library), I noticed that the MQTT client won't reconnect, if the pointer to the last will topic has become invalid. That's why I stored the last will topic in a previously defined variable in the scope of the class. I think we should do it this way for the JSON wrapped topic, too.